controller/appcontroller.go

```diff
@@ -446,6 +448,49 @@ func (ctrl *ApplicationController) getResourceTree(a *appv1.Application, managed
 	return &appv1.ApplicationTree{Nodes: nodes, OrphanedNodes: orphanedNodes, Hosts: hosts}, nil
 }
 
+type PodMetrics struct {
+	Kind       string `json:"kind,omitempty" protobuf:"bytes,1,opt,name=kind"`
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,2,opt,name=apiVersion"`
+	Metadata   struct {
+		Name              string    `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+		Namespace         string    `json:"namespace,omitempty" protobuf:"bytes,3,opt,name=namespace"`
+		SelfLink          string    `json:"selfLink,omitempty" protobuf:"bytes,4,opt,name=selfLink"`
+		CreationTimestamp time.Time `json:"creationTimestamp,omitempty" protobuf:"bytes,8,opt,name=creationTimestamp"`
+	} `json:"metadata"`
+	Timestamp  metav1.Time
+	Window     metav1.Duration
+	Containers []ContainerMetrics
+}
+
+type ContainerMetrics struct {
+	Name  string
+	Usage struct {
+		CPU    string `json:"cpu"`
+		Memory string `json:"memory"`
+	} `json:"usage"`
+}
+
+func getMetric(clientset *kubernetes.Clientset, pod *PodMetrics, podName string, podNamespace string) error {
+	if podNamespace == "" {
+		podNamespace = "default"
+	}
+	data, err := clientset.RESTClient().Get().AbsPath("apis/metrics.k8s.io/v1beta1/namespaces/" + podNamespace + "/pods/" + podName).DoRaw(context.Background())
```

Comment:
Alex: This function is used by getResourceTree method. The getResourceTree get's called every reconciliation and for performance, resources cannot make API called - it access only in-memory cached data. Can we use informer to store metrics?

Hi @alexmt, could you please elaborate on that question? Do we for example store the metrics whenever the informer is updated? It looks like that tree needs to be updated regardless to 'hold' the metric info so the UI can retrieve it. What do you think about populatePodInfo in info and have PodInfo store the metrics?